File: Python/Heap_PriorityQueue_Easy_KthLargestElementInAStream_703.py
class KthLargest:

    def __init__(self, k: int, nums: List[int]):
        self.heapLimit = k
        self.heap = []

        for i in range(0, len(nums)):
            heapq.heappush(self.heap, nums[i])
        

    def add(self, val: int) -> int:

        heapq.heappush(self.heap, val)

        if len(self.heap) < self.heapLimit:
            return -1
        else:
            while len(self.heap) > self.heapLimit:
                heapq.heappop(self.heap)

        return self.heap[0]


# Your KthLargest object will be instantiated and called as such:
# obj = KthLargest(k, nums)
# param_1 = obj.add(val)

File: C++/Heap_PriorityQueue_Easy_KthLargestElementInAStream_703.cpp
class KthLargest {
private:
    int heapLimit;
    priority_queue<int, vector<int>, greater<int>> minHeap;

public:

    KthLargest(int k, vector<int>& nums) 
    {
        heapLimit = k;

        for(int i = 0; i < nums.size(); i++)
        {
            minHeap.push(nums[i]);
        }
    }
    
    int add(int val) 
    {
        minHeap.push(val);

        if(minHeap.size() < heapLimit)
        {
            return -1;
        }
        else 
        {
            while(minHeap.size() > heapLimit)
            {
                minHeap.pop();
            }
        }

        return minHeap.top();
    }
};

/**
 * Your KthLargest object will be instantiated and called as such:
 * KthLargest* obj = new KthLargest(k, nums);
 * int param_1 = obj->add(val);
 */